### PR TITLE
Refactored CalendarEventWithViewAlert, CalendarEventWidgetService

### PR DESCRIPTION
**Pull request by praslnx8**

### CODE REVIEW

File: app/src/main/java/com/mymeetings/android/view/widgets/CalendarEventWidgetService.kt [hunk 1]
```diff
@@ -73,120 +70,72 @@ class CalendarEventWidgetRemoteViewFactory(
...
-    }
+    private fun setDataToRemoteViews(
+        remoteViews: RemoteViews,
+        calendarEventWithAlert: CalendarEventWithAlert
+    ) {
+        val calendarEvent = calendarEventWithAlert.calendarEvent
+        val calendarEventViewAlert = calendarEventWithAlert.eventViewAlert
+
+        val startTimeToEndTimeString =
+            "${clockUtils.getFormattedTime(calendarEvent.startTime)} - ${clockUtils.getFormattedTime(
```
Comment (amitd291) [inline app/src/main/java/com/mymeetings/android/view/widgets/CalendarEventWidgetService.kt:90]: Logical code in UI should be avoided.

File: app/src/main/java/com/mymeetings/android/model/managers/CalendarEventAlertManager.kt [hunk 2]
```diff
@@ -1,55 +1,20 @@
...
-            eventStartTime - clockUtils.currentTimeMillis() < TimeUnit.MINUTES.toMillis(1) -> {
-                "In a minute"
-            }
-            else -> {
-                clockUtils.getTimeLeft(eventStartTime)
-            }
-        }
-    }
+    private fun getViewAlert(calendarEvent: CalendarEvent) = EventViewAlert(
+        timeToShowItAsImmediateEvent = calendarEvent.startTime - priorityTime,
```
Comment (amitdash291) [inline app/src/main/java/com/mymeetings/android/model/managers/CalendarEventAlertManager.kt]: Namings are not clear, can discuss on this.

File: app/src/main/java/com/mymeetings/android/model/EventViewAlert.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.mymeetings.android.model
+
+import com.mymeetings.android.model.strategies.ViewAlertType
+import com.mymeetings.android.utils.ClockUtils
+
+class EventViewAlert(
+    timeToShowItAsImmediateEvent: Long?,
```
Comment (amitd291) [inline app/src/main/java/com/mymeetings/android/model/EventViewAlert.kt]: Namings are not clear, can discuss on this.

File: app/src/main/java/com/mymeetings/android/model/EventViewAlert.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    timeToShowItAsRunningEvent: Long,
+    clockUtils: ClockUtils
```
Comment (amitd291) [inline app/src/main/java/com/mymeetings/android/model/EventViewAlert.kt]: Why is `clockUtils` passed to a model class, is this behaving like a ViewModel?

File: app/src/main/java/com/mymeetings/android/view/widgets/CalendarEventWidgetService.kt [hunk 5]
```diff
@@ -73,120 +70,72 @@ class CalendarEventWidgetRemoteViewFactory(
...
-                    val startTimeToEndTimeString = "${it.startTime} - ${it.endTime}"
-                    if (it.viewAlertType == ViewAlertType.RUNNING) {
-                        setTextViewText(R.id.runningTitleText, it.calendarEvent.title)
-                        setTextViewText(R.id.runningTimeText, startTimeToEndTimeString)
-                        setViewVisibility(R.id.upcomingLayout, View.GONE)
-                        setViewVisibility(R.id.runningLayout, View.VISIBLE)
-                    } else {
-                        setTextViewText(R.id.titleText, it.calendarEvent.title)
-                        setTextViewText(R.id.timeText, startTimeToEndTimeString)
-                        setTextViewText(R.id.timeLeftToStartText, it.timeLeftToStart)
```
Comment (amitd291) [inline app/src/main/java/com/mymeetings/android/view/widgets/CalendarEventWidgetService.kt:88]: This way of accessing String fields like `timeLeftToStart ` in the UI was much readable and clearer than the current approach.

File: app/src/main/java/com/mymeetings/android/model/EventViewAlert.kt (hunk 4 above)
Comment (praslnx8) [reply app/src/main/java/com/mymeetings/android/model/EventViewAlert.kt]: Its a Domain model that deals with ViewAlert of Event. So it has logics. We are going to use ViewModel just for presentation.

**Pull request merged**